## Imports

In [74]:
import pandas as pd
import numpy as np
import datetime, random

from multiprocessing import Pool
import time
import os

## Reading in files; managing columns

In [126]:
%%time
df = pd.read_csv(r'H:\Data\Standardized AKI definition\dataset\output.csv')
df['time'] = pd.to_datetime(df.time)
df = df.drop(['Unnamed: 0'], axis=1)

yu = pd.read_csv(r'H:\Data\Standardized AKI definition\dataset\aki flagger covid creatinines with covidtest.csv')
yu['pat_mrn_id'] = yu.pat_mrn_id.str.strip('MR').astype('int') #indexing with integers is quicker
yu['time'] = pd.to_datetime(yu.time)
yu['covidtest'] = yu.covidtest.astype('bool')
yu.columns = ['mrn', #Renaming columns for ease
              'pat_enc_id',
              'time',
              'creat',
              'enc',
              'running_aki_stage',
              'historical_aki_stage',
              'covid'] 

yu['running_aki'] = yu.running_aki_stage > 0 
yu['historical_aki'] = yu.historical_aki_stage > 0

yu = yu.groupby('enc').apply(lambda d: d[~d.time.duplicated()])
yu = yu.reset_index(drop=True)

print(yu.shape)

(127254, 10)
Wall time: 24.6 s


In [127]:
%%time
df_subset = df.set_index(['mrn', 'enc', 'time']).loc[yu.set_index(['mrn','enc','time']).index]
df_subset = df_subset.reset_index().groupby('enc').apply(lambda d: d[~d.time.duplicated()])
df_subset = df_subset.reset_index(drop=True)
print(np.all(df_subset.time == yu.time))
print(np.all(df_subset.enc == yu.enc))
print(np.all(df_subset.mrn == yu.mrn))
print('Subset shape:',df_subset.shape, 'Yu\'s df shape:',yu.shape)
df_subset.loc[df_subset.rollingwindow_aki.isnull(), 'rollingwindow_aki'] = 0
df_subset.loc[df_subset.backcalc_aki.isnull(), 'backcalc_aki'] = 0

True
True
True
Subset shape: (127254, 17) Yu's df shape: (127254, 10)
Wall time: 34.6 s


### Check mismatch 

###### Back-calc match: 127105 / 127254; 99.88%

###### Rolling-window match: 127215 / 127254; 99.97%

In [128]:
print('Back-calc match:', (df_subset.backcalc_aki == yu.historical_aki).sum(), '/', df_subset.shape[0],
      ';',(df_subset.backcalc_aki == yu.historical_aki).sum() / df_subset.shape[0])
print('Rolling-window match:', (df_subset.rollingwindow_aki == yu.running_aki).sum(), '/', yu.shape[0],
      ';', (df_subset.rollingwindow_aki == yu.running_aki).sum() / yu.shape[0])

Back-calc match: 127105 / 127254 ; 0.9988291134266899
Rolling-window match: 127215 / 127254 ; 0.9996935263331604


### Back-calc mismatch

657313, 4143535

In [144]:
yu.iloc[np.where(df_subset.backcalc_aki != yu.historical_aki)[0]].mrn.values

array([ 657313, 4143535,  197069,  197069,  197069,  197069, 1248799,
       3232988, 3232988, 3232988, 3232988, 5356360, 3542945, 2483162,
       4488514, 6476714, 5548292, 5031590, 2499045, 6408233, 1090433,
       5895234,   53504, 2130749, 4171429, 1177106, 1854532,   69460,
       3670657, 1652965, 5565198, 1729355, 5904941, 1801029, 6428971,
       6428971, 1130875, 1281700, 5955510,  659481, 5901458,   46813,
       1046434, 1870208,  463111,  463111,  463111, 3491149, 5034627,
        113292, 3235213, 5543343, 5963419, 2727979,   33230, 2127519,
         85146, 1836310, 5140831,  385219,  899502, 5450735, 3626664,
       1710520,  658845,  658845,  658845, 1437576, 6024049, 1741452,
       4614648, 3405350, 3385480, 5021392, 4952780, 5153405, 5153405,
       5153405, 5153405, 2075680, 2075680, 6501816, 2815476, 2182049,
       2532499,  445947,  934131, 1757980, 4514643,  566792, 2314043,
       1169533, 1596792, 3063014, 4461359, 1955929, 1374009, 3235213,
       2258170, 3912

In [145]:
#df_subset.iloc[np.where(df_subset.backcalc_aki != yu.historical_aki)[0]]

### Rolling-window mismatch

We mismatch on mrn: 6496928, 6500618, 5446541, ...

In [148]:
#yu.iloc[np.where(df_subset.rollingwindow_aki != yu.running_aki)[0]]

In [149]:
#df_subset.iloc[np.where(df_subset.rollingwindow_aki != yu.running_aki)[0]]

## Reading in other dataframes, trying to make heads and tails of what data is included and what isn't

In [124]:
yu_covid_mrn = pd.read_csv(os.path.join(baseFolder, 'aki flagger covid creatinines with covidtest mrn.csv'))
print(yu_covid_mrn.shape)

yu_covid = pd.read_csv(os.path.join(baseFolder, 'aki flagger covid creatinines with covidtest.csv'))
print(yu_covid.shape)

yu = pd.read_csv(os.path.join(baseFolder, 'aki flagger covid creatinines.csv')) #these were all tested for covid
print(yu.shape)

covid_test_result = pd.read_csv(os.path.join(baseFolder, 'covid test result.csv'))
covid_test_result['mrn'] = covid_test_result.pat_mrn_id.str.strip('MR').astype('int')
covid_test_result['covidtest'] = covid_test_result.covidtest.astype('bool')
print(covid_test_result.shape)

covid_creats = pd.read_csv(os.path.join(baseFolder, 'covid creatinines.csv'))

covid_creats['mrn'] = covid_creats.pat_mrn_id.str.strip('MR').astype('int')
print(covid_creats.shape, df.shape)

(1089, 3)
(127272, 8)
(127272, 6)
(25265, 5)
(299718, 12) (215121, 17)


In [233]:
%%time
indices = [df.mrn[indx] in covid_test_result.mrn.values for indx in range(df.shape[0])]
yu_indices = [yu.mrn[indx] in covid_test_result.mrn.values for indx in range(yu.shape[0])]
cc_indices = [covid_creats.mrn[indx] in yu.mrn.values for indx in range(covid_creats.shape[0])]
df_indices = [df.mrn[indx] in yu.mrn.values for indx in range(df.shape[0])]
df_enc_indices = [df.enc[indx] in yu.enc.values for indx in range(df.shape[0])]
covid_mrn_indices = [df.mrn[indx] in covid_result.mrn.values for indx in range(df.shape[0])]
covid_enc_indices = [df.enc[indx] in covidpos.enc.values for indx in range(df.shape[0])]

Wall time: 1min 25s


In [4]:
%%time
baseFolder = r'H:\Data\Standardized AKI definition\dataset'
covid_result = pd.read_csv(os.path.join(baseFolder, 'covid test result.csv'))
covid_result['mrn'] = covid_result.pat_mrn_id.str.strip('MR').astype('int')
covid_result['enc'] = covid_result.enc_id
covid_result['covidtest'] = covid_result.covidtest.astype('bool')
covid_result['time'] = pd.to_datetime(covid_result.time)
covid_result = covid_result.drop(['pat_enc_csn_id','pat_mrn_id', 'enc_id'], axis=1)

print('covid_result shape:',covid_result.shape, 'df shape:',df.shape,'yu shape:',yu.shape)
print('yu encs:', yu.enc.unique().shape[0], 'yu mrns:', yu.mrn.unique().shape[0])
print('covid+ tests:', covid_result.covidtest.sum())
print('covid encs:',covid_result.enc.unique().shape[0], 'covid mrns:',covid_result.mrn.unique().shape[0])

covid_result shape: (25265, 4) df shape: (295435, 17) yu shape: (127254, 10)
yu encs: 15730 yu mrns: 13821
covid+ tests: 4688
covid encs: 17441 covid mrns: 15454
Wall time: 120 ms
